### PR TITLE
remove random string when unnecessary

### PR DESCRIPTION
**Pull request by vbmithr**

### CODE REVIEW

File: lwt/websocket_lwt.ml [hunk 1]
```diff
@@ -44,7 +44,8 @@ module Connected_client = struct
   let create
       ?read_buf
       ?(write_buf=Buffer.create 128)
-      random_string http_request flow ic oc =
+      ?random_string
```
Comment (copy) [inline lwt/websocket_lwt.ml]: This is always `None`, can be removed from the record.

File: core/websocket.ml [hunk 2]
```diff
@@ -203,19 +203,30 @@ module IO(IO: Cohttp.S.IO) = struct
...
     Buffer.clear buf;
-    write_frame_to_buf ?random_string ~masked buf @@ Frame.close code;
+    write_frame_to_buf ?masked buf @@ Frame.close code;
     write oc @@ Buffer.contents buf
 
-  let make_read_frame ?(buf=Buffer.create 128) ?random_string ~masked ic oc = fun () ->
+  let make_read_frame
+      ?(buf=Buffer.create 128)
+      ~random_string
+      ~masked ic oc = fun () ->
```
Comment (copy) [inline core/websocket.ml]: `masked` should have the same type as `write_frame_to_buf`, `random_string` can be removed then.

File: lwt/websocket_lwt.ml [hunk 3]
```diff
@@ -42,13 +41,12 @@ module Connected_client = struct
   }
 
   let create
+      ?(random_string=Rng.init ())
```
Comment (copy) [inline lwt/websocket_lwt.ml]: Argument can be removed.

**Comment by vbmithr** [comment]
If it's good for you, I'll squash the patches before merge.

**Pull request closed without merging**